app/src/main/java/com/cortado/minionify/accumulators/PreferenceAccumulator.java

```diff
@@ -0,0 +1,37 @@
+package com.cortado.minionify.accumulators;
+
+import android.content.SharedPreferences;
+
+import com.cortado.minionify.accumulators.exceptions.KeyDoesNotExists;
+
+public class PreferenceAccumulator implements Accumulator {
+    private SharedPreferences mPreferences;
+
+    private static final String KEY_NOT_FOUND_DEFAULT_VALUE = "%%!$";
+
+    public PreferenceAccumulator(SharedPreferences prefs) {
+        mPreferences = prefs;
+    }
+
+    @Override
+    public void set(String key, String value) {
+        SharedPreferences.Editor prefsEditor = mPreferences.edit();
+        prefsEditor.putString(key, value);
+        prefsEditor.commit();
+    }
+
+    @Override
+    public String get(String key) {
+        if (!exists(key)) {
+            throw new KeyDoesNotExists();
```

Comment:
declare it in the interface. we usually dont declare runtime exceptions, but when that's a case when you want it to be obvious to the user

Reply:
Done.